### ISSUE TITLE
Panic in `cairo1-run` when printing the output of a Cairo program that initializes a struct from a function

### ISSUE THREAD

**Issue by raphaelDkhn** [lambdaclass/cairo-vm#1670]
**Describe the bug**

Hi, cairo1-run panics when I run `cargo run programs/tensor_fp.sierra --layout all_cairo --print_output` of the following Cairo program:
```cairo
use orion::numbers::{FP16x16, FixedTrait};
use orion::operators::tensor::{Tensor, TensorTrait, FP16x16Tensor};

fn main() -> Tensor<FP16x16> {
     TensorTrait::new(
         array![1, 2].span(),
         array![
           FixedTrait::new(1, false), 
           FixedTrait::new(1, true)
         ].span()
    )
}
>>>
thread 'main' panicked at cairo1-run/src/cairo_run.rs:795:56:
called `Result::unwrap()` on an `Err` value: RelocatableSubDiffIndex((Relocatable { segment_index: 7, offset: 0 }, Relocatable { segment_index: 6, offset: 2 }))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

However when I initialize Tensor and FP16x16 structs without the `new` function as follow it returns the correct output:
```cairo
use orion::numbers::{FP16x16, FixedTrait};
use orion::operators::tensor::{Tensor, TensorTrait, FP16x16Tensor};

fn main() -> Tensor<FP16x16> {
    Tensor {
        data: array![
            FP16x16 { mag: 1, sign: false }, 
            FP16x16 { mag: 2, sign: true },
        ].span(),
        shape: array![1, 2].span()
    }
}
>>>
Program Output : [1 2] [1 false 2 true]
```

**What version/commit are you on?**
I get this issue when I'm using this PR #1665 (cc. @fmoletta)

**Additional context**
Here are the initialized function `new` of [Tensor](https://github.com/gizatechxyz/orion/blob/74e63f34275e257cf21ce13c440b1940bff812ca/src/operators/tensor/core.cairo#L5856-L5859) and [FP16x16](https://github.com/gizatechxyz/orion/blob/74e63f34275e257cf21ce13c440b1940bff812ca/src/numbers/fixed_point/implementations/fp16x16/core.cairo#L39-L41) on Orion lib.

**Comment by raphaelDkhn** [comment]
> Hello! Thanks for reporting this! I just pushed a fix to #1665

Thanks @fmoletta for the follow up! I just tried again the program with the fixes you did and I get that error:
```bash
Error: VirtualMachine(Unexpected)
```

**Comment by raphaelDkhn** [comment]
Thank you @fmoletta for the follow up, it's perfectly working!
